tensorflow/compiler/xla/service/cpu/cpu_runtime_sse4_1.h

```diff
@@ -24,22 +24,33 @@ limitations under the License.
 
 #include "tensorflow/core/platform/macros.h"
 
+// MSVC does not have __SSE4_1__ macro. Eigen enables EIGEN_VECTORIZE_SSE4_1
+// when __AVX__ is defined, we should do the same.
+#if defined(__SSE4_1__) || defined(__AVX__)
```

Comment:
I'd suggest putting all of these definitions in a single header; and have this `defined(__SSE4_1__) || defined(__AVX__)` logic only for MSVC.  I don't think the latter bit makes a functional difference since clang defines `__SSE4_1__` when building with `-mavx`, but it will make the motivation clearer.

I'd also avoid `#define TF_XLA_HAS_SSE4_1 1` and just have `#define TF_XLA_HAS_SSE4_1`.  Even though what you have makes `GetAvailableIntrinsics` shorter, it allows us to mistakenly write `#ifdef TF_XLA_HAS_SSE4_1`.

Reply:
> and have this defined(__SSE4_1__) || defined(__AVX__) logic only for MSVC

Done. Becomes `defined(__SSE4_1__) || (defined(_MSC_VER) && defined(__AVX__))`.

> I'd also avoid #define TF_XLA_HAS_SSE4_1 1 and just have #define TF_XLA_HAS_SSE4_1. Even though what you have makes GetAvailableIntrinsics shorter, it allows us to mistakenly write #ifdef TF_XLA_HAS_SSE4_1.

Done. I did mistakenly write `#ifdef` while working on the old patch, so I can understand your concern.

> I'd suggest putting all of these definitions in a single header

I have eventually decided to not splitting the definitions out into a new header file. That requires adding a new `cc_library` and make three `cc_library(name = "cpu_runtime_*")` to depend on it due to Bazel's strict header dependency. Seems over-complicated.